packages/saved-views-react/src/applySavedView.ts

```diff
@@ -115,6 +115,33 @@ export async function applySavedView(
       viewState.applyPose(viewport.view.savePose());
     }
 
+    viewport.changeView(viewState, settings.viewChangeOptions);
+  } else if (settings.camera !== "keep") {
+    // We use "hidden" as the default value for modelAndCategoryVisibilityFallback
+    // because users expect modelSelector.enabled and categorySelector.enabled to
+    // act as exclusive whitelists when modelSelector.disabled or categorySelector.disabled
+    // arrays are empty, respectively.
+    const { modelAndCategoryVisibilityFallback = "hidden" } = settings;
+    const viewState =
+      viewport.view instanceof ViewState
+        ? viewport.view.clone()
+        : await createViewState(iModel, viewport.view, {
+            modelAndCategoryVisibilityFallback,
+          });
+
+    if (settings.camera instanceof ViewPose) {
+      viewState.applyPose(settings.camera);
+    } else {
+      const newViewState = await createViewState(
+        iModel,
+        savedViewData.viewData,
+        {
+          modelAndCategoryVisibilityFallback,
+        },
+      );
+      viewState.applyPose(newViewState.savePose());
+    }
```

Comment:
Just to make things more readable, does it make sense to maybe handle camera always separately from other view state props? And separate the handling of that into functions? This just feels like we are still coupling the logic of applying those "settings" (given that its an if and else-if depending on different setting values).

Perhaps something like:
```
function applyViewStateOptions(...seedViewState: ViewState): ViewState {
   if (settings.viewState == "apply") {
      // do stuff
   }
   ...
   return viewState;
}

function applyCameraOptions(...): ViewState {
   ...
   return viewState;
}

function applySavedView(...) {
   let viewState = createSeedViewState(...);
   viewState = applyViewStateOptions(...);
   viewState = applyCameraOptions(...);
   ...
}
```

Reply:
Another thought is to not even create the view state yet, but create the view state props *from* the multiple sources. Really what it means for camera to be applied is to set the view definition props: https://www.itwinjs.org/reference/core-common/views/viewdefinition3dprops/

And for the view state here it should really just be related to all other parts of the view state props:
https://www.itwinjs.org/reference/core-common/views/viewstateprops/

Reply:
Okay, finally got this up and working with these changes, separating out viewstate settings from camera settings. :)